### PR TITLE
Add descriptions for components and views

### LINKED ISSUE #218: Add descriptions to the StepByStep mode
### Description  
Getting to:
? Choose an action: 
  Complete & Run 
  ──────────────
❯ Add component 
  ──────────────
  Add view 
 

It is not clear enough what _Component_ and _View_ 

### Steps to reproduce  

1. Add descriptions
2. Remove lines in-between options.

### Expected result  

? Choose an action: 
    Complete & Run 
❯ Add component ........ adds customisable component
    Add view....................adds pre-build applications

### PR DESCRIPTION
**Pull request by IvayloG**
Closes #218  .  

WIP. TODO:
1) check/update all template descriptions to more meaningful or short ones (where applicable).
2) format all components when console viewPort is too narrow. Currently it works for grids/editors/scenarios.
3) fix failing tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ Quickly create projects including [Ignite UI for Angular](https://www.infragisti
 ## Overview
 ### Features:
 - Create project structure
-- Add views with Ignite UI components (e.g. Combo, Grid or Chart)
+- Add scenarios with Ignite UI components (e.g. Combo, Grid or Chart)
```
Comment (damyanpetev) [inline README.md]: nop. (see next bullet)

File: lib/BaseProjectLibrary.ts [hunk 2]
```diff
@@ -119,6 +123,53 @@ export class BaseProjectLibrary implements ProjectLibrary {
...
 	}
+	public getComponentDescriptionByName(name: string): string {
+		return this.components.find(x => x.name === name).templates[0].description;
+	}
+
+	public getCustomTemplates(): Template[] {
+		return this.customTemplates;
+	}
+
+	public formatOutput(templates: Template[]): string[] {
```
Comment (damyanpetev) [inline lib/BaseProjectLibrary.ts]: This one should be part of the prompt session logic (handling templates) and the general formatting should be in Util seeing as the `list` command should be using it too.

File: lib/BaseProjectLibrary.ts [hunk 3]
```diff
@@ -119,6 +123,53 @@ export class BaseProjectLibrary implements ProjectLibrary {
...
+		const result: string[] = [];
+		const addCharsCount = 3;
+		const spaceChar = " ";
+		const truncateChar = ".";
+		let maxNameLength = 0;
+		const viewPort = process.stdout.columns;
+		const cutSuffix = 3;
+		const nameLeftPadding = 3;
+
+		for (const template of templates) {
```
Comment (damyanpetev) [inline lib/BaseProjectLibrary.ts]: `maxNameLength = Math.max(...templates.map(x => x.name.length))`

File: lib/PromptSession.ts [hunk 4]
```diff
@@ -444,17 +458,116 @@ export class PromptSession {
...
+	/**
+	 * Generates a list of options for addComponent
+	 * @param projectLibrary to generate options for
+	 */
+	private generateGroupChoices(projectLibrary: ProjectLibrary): Array<{}> {
+		const groups = projectLibrary.getComponentGroups();
+		const groupChoices: Array<{}> = [];
+		for (const group of groups) {
+			groupChoices.push({
+				name: group + chalk.gray(`...........contains similar ${group} components`),
```
Comment (damyanpetev) [inline lib/PromptSession.ts]: @bazal4o We might need to add an interface for the groups so they can.. have actual descriptions.
That on it's own is a pretty decent chunk of extra changes.. 😒

File: lib/PromptSession.ts [hunk 5]
```diff
@@ -444,17 +458,116 @@ export class PromptSession {
...
+		groupName: string,
+		componentName?: string): Array<{}> {
+
+		const componentChoices: Array<{}> = [];
+		const groupedComponents = projectLibrary.getComponentNamesByGroup(groupName);
+		for (const component of groupedComponents) {
+			const hasChildComponents = projectLibrary.getComponentByName(component).templates.length > 1;
+			if (hasChildComponents) {
+				componentChoices.push({
+					name: component + chalk.gray(`...........choose from ${component} templates`),
```
Comment (damyanpetev) [inline lib/PromptSession.ts]: `Component` should have `description` too, that one at least is easy to add.

File: spec/unit/BaseProjectLibrary-spec.ts [hunk 6]
```diff
@@ -165,6 +165,7 @@ describe("Unit - Base project library ", () => {
 				// tslint:disable-next-line:no-object-literal-type-assertion
 				return {
 					templates: {
+						//components: { name: folder + "ComponentName", description: "ComponentDescription"},
```
Comment (bazal4o) [inline spec/unit/BaseProjectLibrary-spec.ts]: Remove unused code

File: spec/unit/BaseProjectLibrary-spec.ts [hunk 7]
```diff
@@ -178,14 +179,19 @@ describe("Unit - Base project library ", () => {
 
 		const library = new BaseProjectLibrary(__dirname);
 
-		const mockTemplate = library.templates[0];
+		const mockTemplate: Template = library.templates[0];
 		mockTemplate.components = ["newComponent"];
+		// mockTemplate.components = [{
```
Comment (bazal4o) [inline spec/unit/BaseProjectLibrary-spec.ts]: unused code

File: spec/unit/BaseProjectLibrary-spec.ts [hunk 8]
```diff
@@ -178,14 +179,19 @@ describe("Unit - Base project library ", () => {
...
+		// 		description: "ComponentDescription",
+		// 		name: "newComponent"
+		// 	}];
 		mockTemplate.name = "newName";
 		library.registerTemplate(mockTemplate);
 
 		expect(library.getTemplateById("gridTemplate")).toBeTruthy();
 		expect(library.getTemplateByName("newName")).toBeTruthy();
 		expect(library.getComponentByName("newComponent")).toBeTruthy();
+		//expect(library.getComponentByName("newComponent").description).toBe("ComponentDescription");
```
Comment (bazal4o) [inline spec/unit/BaseProjectLibrary-spec.ts]: either uncomment the expect, or remove it

**Pull request merged**